mbedtls/src/cipher/mod.rs

```diff
@@ -290,12 +290,12 @@ impl Cipher<Encryption, Authenticated, AdditionalData> {
     pub fn encrypt_auth_inplace(
         mut self,
         ad: &[u8],
-        data: &mut [u8],
```

Comment:
This is not strictly necessary - the wrapper could compute the tag length, concatenate, and feed `encrypt_auth_inplace`.

This API change (and others) mean that
- Every consumer of this API needs to adapt their code and perform the above. More code changes downstream, more risk.
- On the other hand this is a wrapper and it makes more sense to indeed reflect the new API.

Both are valid points in my opinion, I believe that reflecting the new API (as you're proposing) is fine. Also, it makes more sense to have the ciphertext and tag together at all times, as in the new API.

Reply:
Yes, at first I try to keep api unchanged but found in that way I need a create temporary [u8] for concatenate, which not looks good & may influence performance

Reply:
Added comments to explain the new API